### PR TITLE
Persist last selected tab in stats

### LINKED ISSUE #309: Persist last selected tab
Present the user with the last tab they had selected rather than defaulting it to Insights.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #309 

Persists the last selected "tab" (segment) in stats and defaults it to Insights like before. This tab will be persisted across the sites - so if the user switches sites the last tab will be still selected. Chances are the user will want to come back to the same tab in every site they have.

Needs Review: @jleandroperez 

Do not merge this PR until #315 is merged (this branch is based off of it)

### CODE REVIEW

File: WordPressCom-Stats-iOS/InsightsTableViewController.m [hunk 1]
```diff
@@ -74,11 +80,17 @@ - (void)viewDidLoad {
     self.tableView.backgroundColor = [WPStyleGuide itsEverywhereGrey];
     self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
 
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"WordPressCom-Stats-iOS" ofType:@"bundle"];
+    NSBundle *bundle = [NSBundle bundleWithPath:path];
+    
     [self.tableView registerClass:[StatsTableSectionHeaderView class] forHeaderFooterViewReuseIdentifier:StatsTableSectionHeaderSimpleBorder];
+    [self.tableView registerNib:[UINib nibWithNibName:@"InsightsWrappingTextCell" bundle:bundle] forCellReuseIdentifier:InsightsTableWrappingTextCellIdentifier];
+    [self.tableView registerNib:[UINib nibWithNibName:@"InsightsWrappingTextCell" bundle:bundle] forCellReuseIdentifier:InsightsTableWrappingTextLayoutCellIdentifier];
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/InsightsTableViewController.m]: Sounds familiar! yay!

File: WordPressCom-Stats-iOS/InsightsTableViewController.m [hunk 2]
```diff
@@ -221,12 +241,26 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
...
+        
+        UILabel *label = (UILabel *)[cell.contentView viewWithTag:100];
+        label.attributedText = [self latestPostSummaryAttributedString];
+        label.preferredMaxLayoutWidth = CGRectGetWidth(tableView.bounds) - 46.0f;
+        [cell setNeedsLayout];
+        [cell layoutIfNeeded];
+        
+        CGSize size = [cell.contentView systemLayoutSizeFittingSize:UILayoutFittingCompressedSize];
+        
+        return size.height;
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/InsightsTableViewController.m]: Definitely, sounds familiar!

File: WordPressCom-Stats-iOS/StatsVisits.h [hunk 3]
```diff
@@ -7,8 +7,8 @@
 @property (nonatomic, strong) NSDate *date;
 
 // NSArray of StatsSummary objects
-@property (nonatomic, strong) NSArray *statsData;
-@property (nonatomic, strong) NSDictionary *statsDataByDate;
+@property (nonatomic, strong) NSArray<StatsSummary *> *statsData;
+@property (nonatomic, strong) NSDictionary<NSDate *, StatsSummary *> *statsDataByDate;
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/StatsVisits.h]: *AWESOME*!!

File: WordPressCom-Stats-iOS/WPStatsService.m [hunk 4]
```diff
@@ -179,6 +180,7 @@ - (void)retrieveInsightsStatsWithAllTimeStatsCompletionHandler:(StatsAllTimeComp
     id allTimeData = cacheDictionary[@(StatsSectionInsightsAllTime)];
     id insightsData = cacheDictionary[@(StatsSectionInsightsMostPopular)];
     id todayData = cacheDictionary[@(StatsSectionInsightsTodaysStats)];
+    id latestPostData = cacheDictionary[@(StatsSectionInsightsLatestPostSummary)];
     id commentsAuthorData = cacheDictionary[@(StatsSubSectionCommentsByAuthor)];
     id commentsPostsData = cacheDictionary[@(StatsSubSectionCommentsByPosts)];
     id tagsCategoriesData = cacheDictionary[@(StatsSectionTagsCategories)];
```
Comment (jleandroperez) [inline WordPressCom-Stats-iOS/WPStatsService.m]: I know, not new code, but (suggestion). Maybe Safe Expectations all over the place?

Comment (astralbodies) [inline WordPressCom-Stats-iOS/WPStatsService.m]: In this case I am bastardizing the dictionary by relying upon mixed data coming back. Dangerous, I know!!

**Comment by astralbodies** [comment]
Waiting to merge until #315 is given a shipit

**Comment by roundhill** [comment]
:fireworks:

**Pull request merged**